android/src/main/kotlin/com/aheaditec/freerasp/Extensions.kt

```diff
@@ -17,3 +23,47 @@ internal inline fun runResultCatching(result: MethodChannel.Result, block: () ->
         result.error(err::class.java.name, err.message, null)
     }
 }
+
+/**
+ * Converts a [SuspiciousAppInfo] instance to a [com.aheaditec.freerasp.generated.SuspiciousAppInfo]
+ * instance used by Pigeon package for Flutter.
+ *
+ * @return A new [com.aheaditec.freerasp.generated.SuspiciousAppInfo] object with information from
+ * this [SuspiciousAppInfo].
+ */
+internal fun SuspiciousAppInfo.toPigeon(context: Context): FlutterSuspiciousAppInfo {
+    return FlutterSuspiciousAppInfo(this.packageInfo.toPigeon(context), this.reason)
+}
+
+/**
+ * Converts a [PackageInfo] instance to a [com.aheaditec.freerasp.generated.PackageInfo] instance
+ * used by Pigeon package for Flutter.
+ *
+ * @return A new [com.aheaditec.freerasp.generated.PackageInfo] object with information from
+ * this [PackageInfo].
+ */
+private fun PackageInfo.toPigeon(context: Context): FlutterPackageInfo {
+    return FlutterPackageInfo(
+        packageName = packageName,
+        appName = context.packageManager.getApplicationLabel(applicationInfo) as String,
```

Comment:
applicationInfo is nullable since API34, can you fix this here? (and please validate that gitbook docs take this into accout as well)